### DISCUSSION TITLE
Does threshold limit remove over the limit entries?

### DISCUSSION THREAD

**Discussion by michaelnguyen2021** [owen-it/laravel-auditing#781] [Q&A] [answered]
| Q                 | A
| ----------------- | ---
| Bug?              | maybe
| New Feature?      | no
| Framework         | Laravel
| Framework version | 5.8
| Package version   | 9.0
| PHP version       | 7.2

#### Actual Behaviour

There are 800 entries of the auditable type. I only want to keep around 100, so I set the 

`protected $auditThreshold = 10;`

When I create a new model and update it, it is saved to `audits` table, but the total entries are 801.

Do I have to run some kind of command to prune the entries manually?

#### Expected Behaviour

The audit threshold should be respected and the total entries should be 10, instead of 801.

#### Steps to Reproduce

1. Create 20 entries of the same auditable type

2. Set $auditThreshold to 10 

3. Create and Update a new auditable type model

4. Check database 

#### Possible Solutions

N/A

**Comment by qwe321389yfs8t7huqd89q** [comment, marked as answer]
For anyone looking for an answer, `$auditThreshold` sets limit per model instance. This means when setting protected `$auditThreshold = 10;` there will be 10 audit records for one model instance, check `auditable_type` and `auditable_id` columns:
```
SELECT COUNT(*), audits.auditable_type, audits.auditable_id
FROM audits
where audits.auditable_type = "App\\Models\\User"
GROUP BY audits.auditable_type, audits.auditable_id
```
